Changes between KDE 3.5.10 tag as modified by kdelibs-3.5-patchset-15.tar.bz2
and KDE 3.5 branch r1074156. Only looking at subdirectory kdelibs here.
Logs since r849627 of the branch but this might be wrong.

r849884 | coolo
updated packages
  M kdeui/Makefile.am
  M kdeui/kdepackages.h
r854560 | skilling
Fixing a few old errors
  M kdoctools/customization/pt-BR/user.entities
r858854 | mueller (= kdelibs-3.5.10-post-r858853.diff in Gentoo patchset 15)
prefer the glibc header of inotify support over the linux header, as the latter is broken with newer kernels
  M kio/kio/configure.in.in
  M kio/kio/kdirwatch.cpp
r858971 | mueller (= kdelibs-3.5.10-post-r858853.diff in Gentoo patchset 15)
fix configure again
  M kio/kio/configure.in.in
r859451 | mueller
fix build for older glibc's again
  M kio/kio/kdirwatch.cpp
r859452 | mueller
perform the uid check on a file that exists even with Linux 2.4
  M kinit/start_kdeinit.c
r874968 | orlovich
Merged revision 874965: Fix crash when fixed-layout tables (wow, someone actually uses those?) specify all % width...
  M khtml/rendering/table_layout.cpp
r886258 | neundorf
 -sync partly with cmake.xml from trunk, i.e. add also all the new cmake commands and keywords, but don't change the...
  M kate/data/cmake.xml
r971511 | winterz
Fix a crash that can happen when using KDatePicker on quitting Kontact. patch by Marc Mutz CCMAIL: mutz@kde.org
  M kdeui/kdatetbl.cpp
r975371 | dfaure
gcc-4.4 fix from http://cvs.pld-linux.org/cgi-bin/cvsweb.cgi/packages/kdelibs/kdelibs-gcc44.patch Not that I care...
  M kioslave/ftp/ftp.cc
r987597 | tmcguire
Fix passwords with '!' in them here as well.
  M kdecore/kstringhandler.cpp
r1002164 | fabo
backport CVE-2009-1725 fix
  M khtml/html/htmltokenizer.cpp
r1017229 | lunakl
managed topmenus should not have a strut, otherwise they can affect other windows, I have no idea why I left this...
  M kdeui/kmenubar.cpp
r1018214 | winterz
add so apidox for kdelibs/kdecore are created.
  A kdecore/Mainpage.dox
r1021970 | winterz
backport necessary stuff from trunk for API generation using a modern doxygen.sh
  A doc/common/Doxyfile.global
  M doc/common/README.Doxygen
  A doc/common/block_title_bottom.png
  A doc/common/block_title_mid.png
  A doc/common/block_title_top.png
  M doc/common/doxygen.css
  M doc/common/fdl-license
  M doc/common/fdl-license.html
  M doc/common/fdl-notice.html
  A doc/common/flat.css
  M doc/common/footer.html
  M doc/common/header.html
  M doc/common/kde-default.css
  A doc/common/kde.css
  M doc/common/mainfooter.html
  M doc/common/mainheader.html
  A doc/common/print.css
  A doc/common/tabs.css
  A doc/common/top-kde.jpg
  A doc/common/top-left.jpg
  A doc/common/top-right.jpg
  A doc/common/top-right.png
  A doc/common/top.jpg
r1022000 | winterz
whoopsie, forgot to change the release to 3.5
  M doc/common/header.html
  M doc/common/mainheader.html
r1033549 | winterz
copy and slightly modify some missing Mainpage.dox from trunk to help us generate some more API dox for the KDE3...
  A dnssd/Mainpage.dox
  A kate/Mainpage.dox
  A kconf_update/Mainpage.dox
  A kded/Mainpage.dox
  A kdesu/Mainpage.dox
  A kdoctools/Mainpage.dox
  A khtml/Mainpage.dox
  A kimgio/Mainpage.dox
  A kinit/Mainpage.dox
  A kjs/Mainpage.dox
  A knewstuff/Mainpage.dox
  A kparts/Mainpage.dox
  A kutils/Mainpage.dox
r1042270 | winterz
add home and business post office box fields approved by tokoe
  M kabc/scripts/entrylist
  M kabc/scripts/field.src.cpp
r1052104 | mueller
fix array overrun (CVE-2009-0689)
  M kjs/dtoa.cpp
r1057062 | tmcguire
Support writing empty directories to ZIP files.
  M kio/kio/kzip.cpp
  M kio/kio/kzip.h
r1057078 | tmcguire
Don't assert here, just ignore the entry. This is used for empty directories in ZIP files.
  M kio/kio/karchive.cpp
r1057086 | tmcguire
Use the correct permissions (S_IFDIR) so that KZip doesn't assert when opening the archive. ...
  M kio/kio/kzip.cpp
r1065517 | tmcguire
Backport r978827 by pino from trunk to the 3.5 branch:
  M kio/kio/kzip.cpp
r1074156 | mueller
fix build with gcc 4.5
  M kate/part/katehighlight.cpp


Index: kate/part/katehighlight.cpp
===================================================================
--- kdelibs/kate/part/katehighlight.cpp	 (.../tags/KDE/3.5.10)	(revision 850549)
+++ kdelibs/kate/part/katehighlight.cpp	 (.../branches/KDE/3.5)	(revision 1074156)
@@ -3225,7 +3225,7 @@
   list.append(charAttribute);
 
   KateAttribute* string = new KateAttribute();
-  string->setTextColor(QColor::QColor("#D00"));
+  string->setTextColor(QColor("#D00"));
   string->setSelectedTextColor(Qt::red);
   list.append(string);
 
@@ -3242,9 +3242,9 @@
 
   KateAttribute* alert = new KateAttribute();
   alert->setTextColor(Qt::black);
-  alert->setSelectedTextColor( QColor::QColor("#FCC") );
+  alert->setSelectedTextColor( QColor("#FCC") );
   alert->setBold(true);
-  alert->setBGColor( QColor::QColor("#FCC") );
+  alert->setBGColor( QColor("#FCC") );
   list.append(alert);
 
   KateAttribute* functionAttribute = new KateAttribute();
Index: kate/data/cmake.xml
===================================================================
--- kdelibs/kate/data/cmake.xml	 (.../tags/KDE/3.5.10)	(revision 850549)
+++ kdelibs/kate/data/cmake.xml	 (.../branches/KDE/3.5)	(revision 1074156)
@@ -1,13 +1,34 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE language SYSTEM "language.dtd">
-<!-- Author: Alexander Neundorf <neundorf@kde.org> //-->
-<!-- cmake 2.0.3 syntax highlighting -->
-<!-- dsKeyword, dsDataType, dsDecVal, dsBaseN, dsFloat, dsChar, dsString, dsComment, dsOthers, new in KDE 3.3: dsAlert, dsFunction, dsRegionMarker, dsError -->
-<language name="CMake" version="1.03" kateversion="2.4" section="Other" extensions="CMakeLists.txt;*.cmake;" mimetype="">
+<!--
+  This file is part of KDE's kate project.
+
+  Copyright 2004 Alexander Neundorf (neundorf@kde.org)
+  Copyright 2005 Dominik Haumann (dhdev@gmx.de)
+  Copyright 2007,2008 Matthew Woehlke (mw_triad@users.sourceforge.net)
+
+ **********************************************************************
+ * This library is free software; you can redistribute it and/or      *
+ * modify it under the terms of the GNU Lesser General Public         *
+ * License as published by the Free Software Foundation; either       *
+ * version 2 of the License, or (at your option) any later version.   *
+ *                                                                    *
+ * This library is distributed in the hope that it will be useful,    *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of     *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU  *
+ * Lesser General Public License for more details.                    *
+ *                                                                    *
+ * You should have received a copy of the GNU Lesser General Public   *
+ * License along with this library; if not, write to the              *
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,   *
+ * Boston, MA  02110-1301, USA.                                       *
+ **********************************************************************
+ -->
+<!-- generated for "cmake version 2.6-patch 2" -->
+<language name="CMake" version="1.06" kateversion="2.4" section="Other" extensions="CMakeLists.txt;*.cmake;" mimetype="" author="Alexander Neundorf (neundorf@kde.org)" license="LGPL">
   <highlighting>
 
     <list name = "commands">
-      <item> ABSTRACT_FILES </item>
       <item> ADD_CUSTOM_COMMAND </item>
       <item> ADD_CUSTOM_TARGET </item>
       <item> ADD_DEFINITIONS </item>
@@ -17,18 +38,26 @@
       <item> ADD_SUBDIRECTORY </item>
       <item> ADD_TEST </item>
       <item> AUX_SOURCE_DIRECTORY </item>
+      <item> BREAK </item>
       <item> BUILD_COMMAND </item>
       <item> BUILD_NAME </item>
       <item> CMAKE_MINIMUM_REQUIRED </item>
+      <item> CMAKE_POLICY </item>
       <item> CONFIGURE_FILE </item>
       <item> CREATE_TEST_SOURCELIST </item>
+      <item> DEFINE_PROPERTY </item>
       <item> ELSE </item>
       <item> ELSEIF </item>
+      <item> ENABLE_LANGUAGE </item>
       <item> ENABLE_TESTING </item>
       <item> ENDFOREACH </item>
+      <item> ENDFUNCTION </item>
       <item> ENDIF </item>
       <item> ENDMACRO </item>
+      <item> ENDWHILE </item>
+      <item> EXECUTE_PROCESS </item>
       <item> EXEC_PROGRAM </item>
+      <item> EXPORT </item>
       <item> EXPORT_LIBRARY_DEPENDENCIES </item>
       <item> FILE </item>
       <item> FIND_FILE </item>
@@ -38,11 +67,14 @@
       <item> FIND_PROGRAM </item>
       <item> FLTK_WRAP_UI  </item>
       <item> FOREACH </item>
+      <item> FUNCTION </item>
       <item> GET_CMAKE_PROPERTY </item>
       <item> GET_DIRECTORY_PROPERTY </item>
       <item> GET_FILENAME_COMPONENT </item>
+      <item> GET_PROPERTY </item>
       <item> GET_SOURCE_FILE_PROPERTY </item>
       <item> GET_TARGET_PROPERTY </item>
+      <item> GET_TEST_PROPERTY </item>
       <item> IF </item>
       <item> INCLUDE </item>
       <item> INCLUDE_DIRECTORIES </item>
@@ -61,6 +93,7 @@
       <item> MACRO </item>
       <item> MAKE_DIRECTORY </item>
       <item> MARK_AS_ADVANCED </item>
+      <item> MATH </item>
       <item> MESSAGE </item>
       <item> OPTION </item>
       <item> OUTPUT_REQUIRED_FILES </item>
@@ -69,14 +102,14 @@
       <item> QT_WRAP_UI </item>
       <item> REMOVE </item>
       <item> REMOVE_DEFINITIONS </item>
+      <item> RETURN </item>
       <item> SEPARATE_ARGUMENTS </item>
       <item> SET </item>
       <item> SET_DIRECTORY_PROPERTIES </item>
+      <item> SET_PROPERTY </item>
       <item> SET_SOURCE_FILES_PROPERTIES </item>
       <item> SET_TARGET_PROPERTIES </item>
       <item> SITE_NAME </item>
-      <item> SOURCE_FILES </item>
-      <item> SOURCE_FILES_REMOVE </item>
       <item> SOURCE_GROUP </item>
       <item> STRING </item>
       <item> SUBDIRS </item>
@@ -84,14 +117,14 @@
       <item> TARGET_LINK_LIBRARIES </item>
       <item> TRY_COMPILE </item>
       <item> TRY_RUN </item>
-      <item> USE_MANGLED_MESA </item>
+      <item> UNSET </item>
       <item> UTILITY_SOURCE </item>
       <item> VARIABLE_REQUIRES </item>
+      <item> VARIABLE_WATCH </item>
       <item> VTK_MAKE_INSTANTIATOR </item>
       <item> VTK_WRAP_JAVA </item>
       <item> VTK_WRAP_PYTHON </item>
       <item> VTK_WRAP_TCL </item>
-      <item> WRAP_EXCLUDE_FILES </item>
       <item> WRITE_FILE </item>
     </list>
 
@@ -106,10 +139,13 @@
       <item> ARGS </item>
       <item> ASCII </item>
       <item> BEFORE </item>
+      <item> BRIEF_DOCS </item>
+      <item> BUNDLE </item>
       <item> CACHE </item>
       <item> CACHE_VARIABLES </item>
       <item> CLEAR </item>
       <item> CMAKE_FLAGS </item>
+      <item> CMAKE_FIND_ROOT_PATH_BOTH </item>
       <item> CODE </item>
       <item> COMMAND </item>
       <item> COMMANDS </item>
@@ -118,45 +154,62 @@
       <item> COMPARE </item>
       <item> COMPILE_FLAGS </item>
       <item> COMPONENT </item>
+      <item> COMPONENTS </item>
       <item> CONFIGURATIONS </item>
+      <item> CONFIGURE </item>
       <item> COPYONLY </item>
       <item> DEFINED </item>
       <item> DEFINE_SYMBOL </item>
+      <item> DEFINITION </item>
       <item> DEPENDS </item>
       <item> DESTINATION </item>
+      <item> DIRECTORY </item>
       <item> DIRECTORY_PERMISSIONS </item>
       <item> DOC </item>
+      <item> DOWNLOAD </item>
       <item> EQUAL </item>
       <item> ESCAPE_QUOTES </item>
+      <item> EXACT </item>
       <item> EXCLUDE </item>
       <item> EXCLUDE_FROM_ALL </item>
       <item> EXISTS </item>
-      <item> EXPORT_MACRO </item>
+      <item> EXPORT </item>
+      <item> EXPR </item>
       <item> EXT </item>
       <item> EXTRA_INCLUDE </item>
       <item> FATAL_ERROR </item>
       <item> FILE </item>
       <item> FILES </item>
       <item> FILE_PERMISSIONS </item>
+      <item> FILES_MATCHING </item>
       <item> FORCE </item>
-      <item> FUNCTION </item>
+      <item> FRAMEWORK </item>
+      <item> FULL_DOCS </item>
       <item> GENERATED </item>
+      <item> GET </item>
       <item> GLOB </item>
       <item> GLOB_RECURSE </item>
+      <item> GLOBAL </item>
       <item> GREATER </item>
       <item> GROUP_SIZE </item>
       <item> HEADER_FILE_ONLY </item>
       <item> HEADER_LOCATION </item>
+      <item> HINTS </item>
       <item> IMMEDIATE </item>
       <item> INCLUDES </item>
       <item> INCLUDE_DIRECTORIES </item>
       <item> INCLUDE_INTERNALS </item>
       <item> INCLUDE_REGULAR_EXPRESSION </item>
+      <item> INHERITED </item>
       <item> INTERNAL </item>
+      <item> IS_ABSOLUTE </item>
+      <item> IS_NEWER_THAN </item>
+      <item> LENGTH </item>
       <item> LESS </item>
       <item> LIBRARY </item>
       <item> LINK_DIRECTORIES </item>
       <item> LINK_FLAGS </item>
+      <item> LINK_INTERFACE_LIBRARIES </item>
       <item> LOCATION </item>
       <item> MACOSX_BUNDLE </item>
       <item> MACROS </item>
@@ -167,19 +220,33 @@
       <item> MATCHES </item>
       <item> MODULE </item>
       <item> NAME </item>
+      <item> NAMELINK_ONLY </item>
+      <item> NAMELINK_SKIP </item>
+      <item> NAMES </item>
       <item> NAME_WE </item>
       <item> NOT </item>
       <item> NOTEQUAL </item>
+      <item> NO_CMAKE_PATH </item>
+      <item> NO_CMAKE_ENVIRONMENT_PATH </item>
+      <item> NO_CMAKE_FIND_ROOT_PATH </item>
+      <item> NO_DEFAULT_PATH </item>
+      <item> NO_MODULE </item>
       <item> NO_SYSTEM_PATH </item>
+      <item> NO_SYSTEM_ENVIRONMENT_PATH </item>
       <item> OBJECT_DEPENDS </item>
+      <item> ONLY_CMAKE_FIND_ROOT_PATH </item>
       <item> OPTIONAL </item>
       <item> OR </item>
       <item> OUTPUT </item>
       <item> OUTPUT_VARIABLE </item>
+      <item> PARENT_SCOPE </item>
       <item> PATH </item>
       <item> PATHS </item>
+      <item> PATH_SUFFIXES </item>
       <item> PATTERN </item>
       <item> PERMISSIONS </item>
+      <item> POLICY </item>
+      <item> POP </item>
       <item> POST_BUILD </item>
       <item> POST_INSTALL_SCRIPT </item>
       <item> PREFIX </item>
@@ -187,40 +254,60 @@
       <item> PRE_BUILD </item>
       <item> PRE_INSTALL_SCRIPT </item>
       <item> PRE_LINK </item>
+      <item> PRIVATE_HEADER </item>
       <item> PROGRAM </item>
       <item> PROGRAMS </item>
       <item> PROGRAM_ARGS </item>
       <item> PROPERTIES </item>
+      <item> PUBLIC_HEADER </item>
+      <item> PUSH </item>
       <item> QUIET </item>
+      <item> RANDOM </item>
       <item> RANGE </item>
       <item> READ </item>
       <item> REGEX </item>
       <item> REGULAR_EXPRESSION </item>
+      <item> RELATIVE_PATH </item>
+      <item> REMOVE </item>
+      <item> REMOVE_RECURSE </item>
       <item> RENAME </item>
       <item> REPLACE </item>
       <item> REQUIRED </item>
+      <item> RESOURCE </item>
       <item> RETURN_VALUE </item>
       <item> RUNTIME </item>
       <item> RUNTIME_DIRECTORY </item>
       <item> SCRIPT </item>
       <item> SEND_ERROR </item>
       <item> SHARED </item>
+      <item> SOURCE </item>
       <item> SOURCES </item>
       <item> STATIC </item>
       <item> STATUS </item>
+      <item> STRIP </item>
+      <item> STRINGS </item>
       <item> STREQUAL </item>
       <item> STRGREATER </item>
       <item> STRLESS </item>
       <item> SUFFIX </item>
+      <item> SUBSTRING </item>
       <item> TARGET </item>
       <item> TARGETS </item>
+      <item> TIMEOUT </item>
       <item> TOLOWER </item>
       <item> TOUPPER </item>
+      <item> TO_CMAKE_PATH </item>
+      <item> TO_NATIVE_PATH </item>
       <item> USE_SOURCE_PERMISSIONS </item>
       <item> VAR </item>
+      <item> VARIABLE </item>
       <item> VARIABLES </item>
       <item> VERSION </item>
+      <item> VERSION_LESS </item>
+      <item> VERSION_EQUAL </item>
+      <item> VERSION_GREATER </item>
       <item> WIN32 </item>
+      <item> WORKING_DIRECTORY </item>
       <item> WRAP_EXCLUDE </item>
       <item> WRITE </item>
     </list>
Index: kate/Mainpage.dox
===================================================================
--- kdelibs/kate/Mainpage.dox	 (.../tags/KDE/3.5.10)	(revision 0)
+++ kdelibs/kate/Mainpage.dox	 (.../branches/KDE/3.5)	(revision 1074156)
@@ -0,0 +1,36 @@
+/** @mainpage Kate Editor Component
+
+The Kate Editor Component (also called Kate Part) implements the
+KTextEditor interfaces. Thus, all Kate Part classes are internal
+and never appear in public interfaces.
+
+If you wish to have a text editor component in your application,
+use the KTextEditor interfaces.
+
+@authors
+Christoph Cullmann \<cullmann@kde.org\><br>
+Joseph Wenninger \<jowenn@kde.org\><br>
+Anders Lund \<anders@alweb.dk\><br>
+Hamish Rodda \<rodda@kde.org\><br>
+Waldo Bastian \<bastian@kde.org\><br>
+Charles Samuels \<charles@kde.org\><br>
+Michael Bartl \<michael.bartl1@chello.at\><br>
+Matt Newell \<newellm@proaxis.com\><br>
+Michael McCallum \<gholam@xtra.co.nz\><br>
+John Firebaugh \<jfirebaugh@kde.org\><br>
+Nadeem Hasan \<nhasan@nadmm.com\><br>
+Jochen Wilhelmy \<digisnap@cs.tu-berlin.de\><br>
+Glen Parker \<glenebob@nwlink.com\><br>
+Michael Koch \<koch@kde.org\><br>
+Dominik Haumann \<dhdev@gmx.de\>
+
+@maintainers
+Christoph Cullmann \<cullmann@kde.org\>
+
+@licenses
+@lgpl
+
+*/
+// DOXYGEN_REFERENCES = kdecore kio kdeui kparts interfaces/kdocument interfaces/ktexteditor kjs
+// DOXYGEN_SET_PROJECT_NAME = Kate
+// vim:ts=4:sw=4:expandtab:filetype=doxygen
Index: kconf_update/Mainpage.dox
===================================================================
--- kdelibs/kconf_update/Mainpage.dox	 (.../tags/KDE/3.5.10)	(revision 0)
+++ kdelibs/kconf_update/Mainpage.dox	 (.../branches/KDE/3.5)	(revision 1074156)
@@ -0,0 +1,31 @@
+/** @mainpage ./kconf_update
+
+kconf_update is a tool designed to update config files. Over time applications
+sometimes need to rearrange the way configuration options are stored. Since
+such an update shouldn't influence the configuration options that the user
+has selected, the application must take care that the options stored in the
+old way will still be honored.
+
+What used to happen is that the application looks up both the old and the
+new configuration option and then decides which one to use. This method has 
+several drawbacks:
+- The application may need to read more configuration files than strictly
+  needed, resulting in a slower startup.
+- The application becomes bigger with code that will only be used once.
+
+kconf_update addresses these problems by offering a framework to update 
+configuration files without adding code to the application itself.
+
+See the <a href="http://websvn.kde.org/trunk/KDE/kdelibs/kconf_update/README.kconf_update?view=markup">README file</a> for more information.
+
+@authors
+Waldo Bastian \<bastian@kde.org\>
+
+@maintainers
+[Unknown/None]
+
+@licenses
+@lgpl
+
+*/
+// vim:ts=4:sw=4:expandtab:filetype=doxygen
